client/packages/cli_new/src/commands/submit/headers.ts

```diff
@@ -0,0 +1,196 @@
+import ora from "ora"
+import { SingleBar, Presets } from "cli-progress"
+import { getConfig } from "@/utils/config.ts"
+import { colorLogMsg, log } from "@/utils/log.ts"
+import { createCircuitContext } from "@/utils/circuit.ts"
+import { Gateway } from "@/schemas/setup.ts"
+import { ApiPromise, WsProvider } from "@polkadot/api"
+import { Circuit } from "@/types.ts"
+import { Encodings } from "@t3rn/sdk"
+
+export const spinner = ora()
+export const progressBar = new SingleBar({}, Presets.shades_classic)
+
+export const handleSubmitHeadersCmd = async (gateway_id: string) => {
+  const config = getConfig()
+  if (!config) {
+    process.exit(1)
+  }
+
+  const gateway = config.gateways.find((g) => g.id === gateway_id)
+  if (!gateway) {
+    log("ERROR", `Gateway with id ${gateway_id} not found in config`)
+    process.exit(1)
+  }
+
+  if (gateway.registrationData.parachain) {
+    log("ERROR", "Headers can only be submitted for a relaychain not parachain")
+    process.exit(1)
+  }
+
+  const { circuit, sdk } = await createCircuitContext()
+
+  try {
+    const transactionArguments = await getHeaders(circuit, gateway, gateway_id)
+
+    spinner.start(`Submitting headers for ${gateway_id}`)
+
+    const tx = sdk.circuit.tx.createBatch(
+      transactionArguments.map((args) => {
+        return circuit.tx.rococoBridge.submitHeaders(
```

Comment:
Here we should match the `verificationVendor` to the correct pallet instance. Currently we have `rococoBridge`, `polkadotBridge`, `kusamaBridge` implemented

Reply:
I have updated the implementation; please refer to commit [0ca4e5c](https://github.com/t3rn/t3rn/pull/881/commits/0ca4e5cb9dd195a81919dec1d05713b9235a0050).